toffy/streak_detection.py

```diff
@@ -0,0 +1,429 @@
+        """
+        fields = [
+            "streak_mask",
+            "streak_df",
+            "filtered_streak_mask",
+            "filtered_streak_df",
+            "boxed_streaks",
+            "corrected_streak_mask",
+        ]
+        if all_data:
+            for field in fields:
+                self._save(name=field)
+        elif name not in fields:
+            raise ValueError(
+                "Invalid field to save. Options include:\
+                \n streak_mask\
+                \n streak_df\
+                \n filtered_streak_mask\
+                \n filtered_streak_df\
+                \n boxed_streaks\
+                \n corrected_streak_mask"
+            )
+        elif name is not None:
+            self._save(name)
+        else:
+            raise ValueError("Please enter a field name, or set `all_data` to True.")
+
+    def _save_dir(self, data_dir: Path, name: str, ext: str):
+        return Path(data_dir, name + f".{ext}")
+    
+    def _save(self, name: str):
+        """Helper function for saving tiff binary masks and dataframes.
+
+        Args:
+            name (str): The field of the DataClass to be saved.
+        """
+        data_dir = Path(self.corrected_dir, f"streak_data_{self.streak_channel}")
+        if not os.path.exists(data_dir):
+            os.makedirs(data_dir)
+
+        data = getattr(self, name)
+        st = partial(self._save_dir, data_dir, name)
+
+        if type(data) is np.ndarray:
+            with TiffWriter(st("tiff")) as tiff:
+                tiff.write(data)
+        elif type(data) is pd.DataFrame:
+            data.to_csv(st("csv"))
+
+
+def _binary_mask(
+    channel: np.ndarray,
+    gaussian_sigma: float = 40,
+    gamma: float = 3.80,
+    gamma_gain: float = 0.10,
+    log_gain: float = 1.00,
+    pmin: int = 80,
+    pmax: int = 95,
+) -> np.ndarray:
+    """Performs a series of denoiseing, filtering, and exposure adjustments to create a binary
+    mask for the given channel.
+
+    Args:
+        channel (np.ndarray): The channel to perform the streak masking on.
+        gaussian_sigma (float, optional): Parameter for `skimage.filters.gaussian`. Defaults to
+        40.
+        gamma (float, optional): Parameter for `skimage.exposure.adjust_gamma`. Defaults to 3.80.
+        gamma_gain (float, optional): Parameter for `skimage.exposure.adjust_gamma`. Defaults to
+        0.10.
+        log_gain (float, optional): Parameter for `skimage.exposure.adjust_log`. Defaults to 1.00.
+        pmin (int, optional): Lower bound for the `np.percentile` threshold, used for rescaling
+        the intensity. Defaults to 80.
+        pmax (int, optional): Upper bound for the `np.percentile` threshold, used for rescaling
+        the intensity. Defaults to 95.
+
+    Returns:
+        np.ndarray: The binary mask containing all of the candidate strokes.
+    """
+    if channel is not None:
```

Comment:
Is there a reason channel would ever be `None`?

Reply:
I don't believe so.